src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,82 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SEPARATOR = ",";
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        writeToFile(createReport(stringDataReport(fromFileName)),
+                new File(toFileName));
+    }
+
+    private String[] stringDataReport(String fromFileName) {
+        StringBuilder stringBuilder = new StringBuilder();
+        String value;
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            while ((value = bufferedReader.readLine()) != null) {
+                stringBuilder.append(value).append(System.lineSeparator());
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file" + fromFileName, e);
+        }
+        return stringBuilder.toString().split(System.lineSeparator());
+    }
+
+    private StringBuilder createReport(String[] text) {
+        int supply = 0;
+        int buy = 0;
+        int result;
```

Comment:
It's a bad practice to declare the variable not in place where you are going to use it